### ISSUE TITLE
Cannot convert {class@00000000000 to class} when constructing AVM2 side of movie clip of type GAME | Event Dispatch - Parameter child must be non-null

### ISSUE THREAD

**Issue by React1-X** [ruffle-rs/ruffle#16504]
### Describe the bug

This issue is present on both desktop and web. The game loads with a white screen, and the console output can be seen in the attached screenshot. Tested on Ruffle Nightly commit ref: 381b0897c92084a120792d7f878058503f9222a2

![image](https://github.com/ruffle-rs/ruffle/assets/157532981/5bd92614-0265-4551-8add-5f8b8b0ab739)

### Expected behavior

The game to load correctly.

### Content Location

The SWF can be found [here](https://github.com/bym-refitted/backyard-monsters-refitted/releases/download/v0.2.6-alpha/bymr-http-stable-v0.2.6-alpha.swf).

The source for the offending file(s) can be found here:

- [GAME.as](https://github.com/bym-refitted/backyard-monsters-refitted/blob/main/client/scripts/GAME.as)
- [AuthForm.as](https://github.com/bym-refitted/backyard-monsters-refitted/blob/main/client/scripts/com/auth/AuthForm.as)

### Affected platform

Online demo

### Operating system

Windows 10

**Comment by Lord-McSweeney** [comment]
The SWF includes two classes with the same name; two lookups find the same class in FP, but find different ones in Ruffle.

**Comment by React1-X** [comment]
> The SWF includes two classes with the same name (`com.auth::AuthForm`); two lookups find the same class in FP, but find different ones in Ruffle.

Hey @Lord-McSweeney,

Apologies, could you elaborate on your comment? What do you mean by there are two classes with the same name?

This is the latest client codebase with one class called `AuthForm.as`:
https://github.com/tram98/backyard-monsters-refitted/tree/main/client/scripts

I appreciate the time. Thank you.

**Comment by Lord-McSweeney** [comment]
For some reason the compiled code ends up with two AuthForm classes in two scripts with the same name, according to @adrian17's investigations ([on our Discord](https://discord.com/channels/610531541889581066/1018988594259574845/1247585345059426335)).

**Comment by React1-X** [comment]
> For some reason the compiled code ends up with two AuthForm classes in two scripts with the same name, according to [@adrian17](https://github.com/adrian17)'s investigations ([on our Discord](https://discord.com/channels/610531541889581066/1018988594259574845/1247585345059426335)).

Thanks very much for the explanation. This is very odd, I'll try on my end to get to the bottom of this; JPEXS making me lose sleep challenge.

**Comment by React1-X** [comment]
Update from my end:

I managed to fix this by deleting the entire class in JPEXS, and then just simply adding it again - I had to delete the same class 12 times, I have no idea how this happened (possibly a bug in the compilation process in old versions of JPEXS).

TLDR; Fixed on my end, but obviously FP was able to run it without an issue even with the problem present.
